File: jbake-core/src/main/java/org/jbake/template/model/DataExtractor.java
package org.jbake.template.model;

import org.jbake.app.ContentStore;
import org.jbake.template.ModelExtractor;
import org.jbake.util.DataFileUtil;
import java.util.Map;
import static org.jbake.app.configuration.PropertyList.*;

public class DataExtractor implements ModelExtractor<DataFileUtil> {

    @Override
    public DataFileUtil get(ContentStore db, Map model, String key) {
        Map<String, Object> config = (Map<String, Object>) model.get("config");
        String defaultDocType = config.get(DATA_FILE_DOCTYPE.getKey().replace(".", "_")).toString();
        DataFileUtil dataUtil = new DataFileUtil(db, defaultDocType);
        return dataUtil;
    }

}

File: jbake-core/src/main/java/org/jbake/template/model/DataExtractor.kt
package org.jbake.template.model

import org.jbake.app.configuration.PropertyList.DATA_FILE_DOCTYPE
import org.jbake.template.TypedModelExtractor
import org.jbake.util.DataFileUtil

class DataExtractor : TypedModelExtractor<DataFileUtil> {

    @Suppress("DEPRECATION", "UNCHECKED_CAST")
    override fun extract(context: RenderContext, key: String): DataFileUtil {
        val model = context.toLegacyMap()
        val config = model["config"] as? Map<String, Any>
        val defaultDocType: String = config?.get(DATA_FILE_DOCTYPE.key.replace(".", "_"))?.toString() ?: ""
        return DataFileUtil(context.db, defaultDocType)
    }
}
